app/screens/edit_server/index.tsx

```diff
@@ -82,14 +101,53 @@ const EditServer = ({closeButtonId, componentId, server, theme}: ServerProps) =>
         }
 
         await DatabaseManager.updateServerDisplayName(server.url, displayName);
+
+        // Update shared password only if it was modified
+        if (preauthSecretModified) {
+            const credentials = await getServerCredentials(server.url);
+            if (credentials) {
+                // Empty field = remove password, otherwise use the new value
+                const newPreauthSecret = preauthSecret.trim() || undefined;
+                setServerCredentials(server.url, credentials.token, newPreauthSecret);
+
+                // Update active REST client
+                try {
+                    const activeClient = NetworkManager.getClient(server.url);
+                    activeClient.setClientCredentials(credentials.token, newPreauthSecret);
+                } catch (error) {
+                    logWarning('Failed to update REST client shared password:', error);
+                }
+
+                // Update active WebSocket client
+                try {
+                    WebSocketManager.createClient(server.url, credentials.token, newPreauthSecret);
+                } catch (error) {
+                    logWarning('Failed to update WebSocket client shared password:', error);
+                }
+            }
+        }
+
         dismissModal({componentId});
-    }, [!buttonDisabled && displayName, !buttonDisabled && displayNameError]);
+    }, [!buttonDisabled && displayName, !buttonDisabled && displayNameError, !buttonDisabled && preauthSecretModified, preauthSecret]);
 
     const handleDisplayNameTextChanged = useCallback((text: string) => {
         setDisplayName(text);
         setDisplayNameError(undefined);
     }, []);
 
+    const handlePreauthSecretTextChanged = useCallback((text: string) => {
+        setPreauthSecret(text);
+        setPreauthSecretModified(true);
+    }, []);
+
+    const handlePreauthSecretFocus = useCallback(() => {
+        // Clear field when focused if it hasn't been modified yet
+        if (!preauthSecretModified) {
+            setPreauthSecret('');
+            setPreauthSecretModified(true); // Mark as modified when clearing dummy value
+        }
```

Comment:
The moment we focus on the field, even if by mistake, we are removing it? Sounds "dangerous". But 0/5.